### PR TITLE
DevTools: Adding support for regex filter

### PR DESCRIPTION
**Pull request by pr8x**
## What does the pull request do?
Adds support for regex filtering in DevTools

image

image

### CODE REVIEW

**Comment by danwalmsley** [comment]
@pr8x if you can resolve the conflicts we can probably merge this.

**Comment by workgroupengineering** [comment]
Great feature. It would be even better if there was an option to save the regex for or reuse it.

File: src/Avalonia.Diagnostics/Diagnostics/ViewModels/TreePageViewModel.cs [hunk 1]
```diff
@@ -54,6 +54,63 @@ private set
...
+                {
+                    ErrorsChanged?.Invoke(this, new DataErrorsChangedEventArgs(nameof(PropertyFilter)));
+                }
+            }
+
+            if (UseRegexFilter)
+            {
+                try
+                {
+                    FilterRegex = new Regex(PropertyFilter);
```
Comment (workgroupengineering) [inline src/Avalonia.Diagnostics/Diagnostics/ViewModels/TreePageViewModel.cs:73]: Wouldn't it be better to compile the regex to improve performance?

**Comment by pr8x** [comment]
I guess we could have a feature to serialize and load the old search query. This should be a separate PR though.

**Comment by pr8x** [comment]
So can we merge this 😅 ?

**Review by grokys — APPROVED** [review]

**Comment by grokys** [comment]
Yeah let's get it merged!

**Pull request merged**